tests/commands/base/test_check_obsolete_data_dir.py

```diff
@@ -0,0 +1,96 @@
+from unittest.mock import MagicMock
+
+import pytest
+
+from briefcase.exceptions import BriefcaseCommandError
+
+from .conftest import DummyCommand
+
+
+def test_skip_if_dot_briefcase_nonexistent(capsys, tmp_path):
+    """Obsolete data dir check does not run if .briefcase dir does not
+    exist."""
+
+    home_path = tmp_path / "home"
+    dot_briefcase_dir = home_path / ".briefcase"
+    cmd = DummyCommand(base_path=tmp_path / "base", home_path=home_path)
+    cmd.check_obsolete_data_dir()
+
+    assert not dot_briefcase_dir.exists()
+    assert capsys.readouterr().out == ""
+
+
+def test_first_notice_if_dot_briefcase_exists(capsys, tmp_path):
+    """Obsolete data dir check shows full notice of transition if .briefcase
+    directory exists but new data directory does not exist."""
+    home_path = tmp_path / "home"
+    dot_briefcase_dir = home_path / ".briefcase"
+    dot_briefcase_dir.mkdir(parents=True)
+
+    cmd = DummyCommand(base_path=tmp_path / "base", home_path=home_path)
+    cmd.data_path = tmp_path / "data_dir"
+    cmd.input.boolean_input = MagicMock()
+    cmd.input.boolean_input.return_value = True
+
+    cmd.check_obsolete_data_dir()
+
+    assert dot_briefcase_dir.exists()
+    assert not cmd.data_path.exists()
+    assert "Briefcase's data directory is changing" in capsys.readouterr().out
+
+
+def test_subsequent_notice_if_dot_briefcase_exists(capsys, tmp_path):
+    """Obsolete data dir check shows notice of transition if .briefcase
+    directory exists and new data directory exists."""
+    home_path = tmp_path / "home"
+    dot_briefcase_dir = home_path / ".briefcase"
+    dot_briefcase_dir.mkdir(parents=True)
+
+    cmd = DummyCommand(base_path=tmp_path / "base", home_path=home_path)
+    cmd.data_path = tmp_path / "data_dir"
+    cmd.data_path.mkdir(parents=True)
+    cmd.input.boolean_input = MagicMock()
+    cmd.input.boolean_input.return_value = True
```

Comment:
This won't be called - we should add a check to verify there won't be any user input requested.

Reply:
Added check that `boolean_input()` is not called.